dali/kernels/audio/mel_scale/mel_filter_bank_cpu.cc

```diff
@@ -39,43 +39,51 @@ namespace audio {
 // For every FFT bin we compute the weight for each filter and travel through the row, computing
 // the contributions on every window of the spectrogram (horizontal axis)
 //
-template <typename T, int Dims>
-class MelFilterBankCpu<T, Dims>::Impl: public MelFilterImplBase<T, Dims> {
+template <typename T>
+class MelFilterBankCpu<T>::Impl: public MelFilterImplBase<T> {
  public:
   template <typename MelScale>
   Impl(MelScale mel_scale, const MelFilterBankArgs &args)
-  : MelFilterImplBase<T, Dims>(mel_scale, args) {
-    intervals_.resize(fftbin_size_, -1);
-    double mel = mel_low_ + mel_delta_;
+      : MelFilterImplBase<T>(mel_scale, args) {
+    auto nfilter = args.nfilter;
 
-    int64_t fftbin = fftbin_start_;
+    intervals_.resize(fftbin_size_, -1);
+    int fftbin = fftbin_start_;
     double f = fftbin * hz_step_;
-
-    int last_interval = args_.nfilter;
-    for (int64_t interval = 0; interval <= last_interval; interval++, mel += mel_delta_) {
-      if (interval == last_interval) {
-        mel = mel_high_;
-      }
-      double freq = mel_scale.mel_to_hz(mel);
+    double mel = mel_low_ + mel_delta_;
+    for (int interval = 0; interval < nfilter + 1; interval++, mel += mel_delta_) {
+      double freq = mel_scale.mel_to_hz(interval == nfilter ? mel_high_ : mel);
       for (; fftbin <= fftbin_end_ && f < freq; fftbin++, f = fftbin * hz_step_) {
         intervals_[fftbin] = interval;
       }
     }
-  }
 
-  void Compute(T* out, const T* in, int64_t nwindows,
-               int64_t out_stride = -1, int64_t in_stride = -1) {
-    if (out_stride <= 0)
-      out_stride = nwindows;
+    mel = mel_low_ + mel_delta_;
+    interval_ends_.resize(nfilter + 2);
+    interval_ends_[0] = fftbin_start_;
+    interval_ends_[nfilter + 1] = fftbin_end_ + 1;
+    for (int interval = 1; interval < nfilter + 1; interval++, mel += mel_delta_) {
```

Comment:
Can this loop be merged with previous one?

Reply:
It could be, but I see no reason to do it.  know it's initialization only, but the inner loop would likely break the optimizations in the inner one (as the compiler would deem it less frequent).